librz/util/path.c

```diff
@@ -180,3 +198,29 @@ RZ_API RZ_OWN char *rz_path_home_expand(RZ_NULLABLE const char *path) {
 
 	return rz_path_home(path + 1);
 }
+
+/**
+ * \brief Return a canonicalized absolute path. Expands all symbolic links and resolves
+ * references to /./, /../ and extra '/' characters.
+ *
+ * \param path Original file path.
+ * \return New canonicalized absolute path.
+ */
+RZ_API RZ_OWN char *rz_path_realpath(RZ_NULLABLE const char *path) {
+#if __UNIX__
+	char buf[PATH_MAX] = { 0 };
+	const char *rp = realpath(path, buf);
+	if (rp) {
+		return strdup(rp);
+	}
+#elif __WINDOWS__
+	char buf[MAX_PATH] = { 0 };
+	char *basename;
+
+	DWORD len = GetFullPathName(path, MAX_PATH, buf, &basename);
```

Comment:
Use `GetFullPathNameW()` and use `rz_utf8_to_utf16()` and `rz_utf16_to_utf8()` to convert from `char *` -> `wchar_t *`->`char *`, currently it will break if `UNICODE` is defined, and its better to always support Unicode paths, whether `UNICODE` is defined or not.